narwhals/_pandas_like/dataframe.py

```diff
@@ -58,7 +58,7 @@
 from narwhals.typing import CompliantDataFrame
 from narwhals.typing import CompliantLazyFrame
 
-CLASSICAL_NUMPY_DTYPES = frozenset(
+CLASSICAL_NUMPY_DTYPES: frozenset[np.dtype] = frozenset(
```

Comment:
You might be able to get away with this?

Possibly an issue for `3.8` though
```suggestion
CLASSICAL_NUMPY_DTYPES = frozenset["np.dtype"](
```